core/src/main/java/org/apache/iceberg/avro/BuildAvroProjection.java

```diff
@@ -49,6 +49,11 @@
     this.current = expectedSchema.asStruct();
   }
 
+  BuildAvroProjection(org.apache.iceberg.types.Type expectedType, Map<String, String> renames) {
```

Comment:
Is this for testing? What guarantees that the expected type is correct?

Reply:
This is for testing only.